src/gflownet/data/replay_buffer.py

```diff
@@ -42,3 +47,15 @@ def sample(self, batch_size):
 
     def __len__(self):
         return len(self.buffer)
+
+
+def detach_and_cpu(x):
+    if isinstance(x, torch.Tensor):
+        x = x.detach().cpu()
+    elif isinstance(x, dict):
+        for k in x.keys():
+            x[k] = detach_and_cpu(x[k])
+    elif isinstance(x, list):
```

Comment:
Could do `isinstance(x, (list, tuple))`.

Reply:
Yes exactly, now removed the cast since we added the tuple case.